### ISSUE TITLE
Support for last.fm scrobbling

### ISSUE THREAD

**Issue by gentlenight479** [ungive/discord-music-presence#53]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the problem

i currently have to use another program for scrobbling and it takes a lot of network usage its not writtern properly and is less active and it only supports apple music for scrobbling and no other media players or services

### Solution

it would be a great feature for last.fm scrobbling, there are no all in one  scrobbler for windows and this app supports  a lot of streaming services and media players 
this will be the one stop for all the scrobbling activites for people who use offline players and streaming services on windows

**Comment by ungive** [comment]
This is definitely on the horizon, might take a bit until I start implementing it though. I'll let you know once the feature arrives!

**Comment by gentlenight479** [comment]
thanks for the reply 
will be waiting

**Comment by hans-juergen** [comment]
What app do you use for scrobbling now?

**Comment by gentlenight479** [comment]
https://github.com/PKBeam/AMWin-RP/issues/116
this
uses a lot of network for me

**Comment by hans-juergen** [comment]
I see, Jonas has joined the #help-api-tools channel on the unofficial Last.fm server today asking for opinions about this idea, by the way:
https://discord.com/channels/260438376270921729/579673026526969876/1294987216770170907

**Comment by ungive** [comment]
I've created roles and a discussion channel where I ask questions on our Discord, feel free to join: https://discord.gg/musicpresence

I'll post a digest of what was discussed here too, once I've figured out the small details, so anyone who's subscribed to this issue gets notifications too, even when they're not in the Discord.

**Comment by RegorForgot** [comment]
Just a request for this future:
Would be really grateful if there was an option for, for example Apple Music, to get rid of the " - EP" or " - Single" or " (feat. [xyz])" at the end of albums and/or tracks? Maybe using Regex?

**Comment by myysty** [comment]
I agree with @RegorForgot's comment, it would also be cool to also remove the & if a track contains multiple primary artists so it would scrobble only the first primary artist

**Comment by RoverBot-Dev** [comment]
any updates on this feature 👀

**Comment by ungive** [comment]
> any updates on this feature 👀

Most certainly! You will see news about scrobbling and the future of Music Presence in the app (and also here on GitHub) this weekend. Stay tuned!

**Comment by RoverBot-Dev** [comment]
it would be nice if we could also use regex to find and replace substrings in metadata for scrobbling
for example: removing everything after & in artist name using custom regex to get only primary artist

**Comment by ungive** [comment]
Hey, just a quick update as promised:

I will publish the news next weekend because I already published news this week and I don't want to be spammy. You can get an idea what it will be about on the Music Presence website: [https://musicpresence.app](https://musicpresence.app?utm_source=github.com&utm_medium=issue-53). The gist is that it's hard for me to find time for adding scrobbling in just my free-time. Working on Music Presence full-time from just donations is a bit of a stretch, but I'm more than ready to commit to the future of Music Presence should I get sufficient funding from users. Right now it's still gonna be a bit until scrobbling will be here because I can't make it my top priority. I hope this helps!

**Comment by ungive** [comment]
I've compiled an implementation plan for how I envision the first working version of scrobbling to look like: #365 

I would love some feedback on it! To summarize: The first version will scrobble to last.fm, allow you to replace metadata fields using regular expressions, cross-reference regular expression matches (e.g. use a match from the title field in the artist field), filter out media that matches regular expressions, toggle scrobbling globally and per media player, and more.

You can follow the progress in that issue. I will soon move it from "Pending" to "In progress", when I start working on it.

**Comment by ungive** [comment]
For anyone subscribed to this issue, scrobbling is in **beta**!

See my comment here for more information: https://github.com/ungive/discord-music-presence/issues/365#issuecomment-3595381416

Some screenshots:

<img width="1472" height="1089" alt="Image" src="https://github.com/user-attachments/assets/fd0f4f76-d71d-46f4-aa42-135f45528e57" />

<img width="853" height="697" alt="Image" src="https://github.com/user-attachments/assets/0190ba65-301d-4822-a1dc-908040390b57" />

**Comment by ungive** [comment]
The scrobbling implementation is feature-complete with scrobbling from any media player, reliable primary artist scrobbling, metadata editing with regular expressions, blacklisting of track/artists/albums and filtering with regular expressions.

For more screenshots see my comment here: https://github.com/ungive/discord-music-presence/issues/365#issuecomment-4149731993. Also check the latest announcement on the [Discord server](https://discord.com/invite/7rc8dWD4ug)!

The next beta will be available tomorrow on the Discord server and I'll open plenty more spots.

<img width="1021" height="1044" alt="Image" src="https://github.com/user-attachments/assets/3dd3c53f-a0a3-4f98-8207-13d8643946ec" />
